### PR TITLE
Streaming population

### PR DESCRIPTION
**Pull request by Janekdererste**
One reviewr suggested weak scaling for our paper. To have this I want to run a 100% sample, which is too big to load it all at once. So, add streaming support for the population, which enables adding persons to the population selectively. 

Please have a look @paulheinr

### CODE REVIEW

File: tests/test_3_links.rs [hunk 1]
```diff
@@ -1,11 +1,33 @@
...
 use rust_q_sim::simulation::messaging::communication::communicators::DummySimCommunicator;
+use rust_q_sim::simulation::network::global_network::Network;
+use rust_q_sim::simulation::population::population::Population;
+use rust_q_sim::simulation::vehicles::garage::Garage;
+
+use crate::test_simulation::{execute_sim, execute_sim_with_channels, TestSubscriber};
 
 mod test_simulation;
 
+fn create_resources(out_dir: &PathBuf) {
```
Comment (paulheinr) [inline tests/test_3_links.rs:14]: Thanks for implementing that!

File: src/simulation/wire_types/population.proto [hunk 2]
```diff
@@ -1,8 +1,9 @@
 syntax = "proto3";
 package population;
 
-message Population {
-  repeated Person persons = 1;
+message Header {
```
Comment (paulheinr) [inline src/simulation/wire_types/population.proto:4]: I didn't really get why we need this header.

**Review by paulheinr — APPROVED** [review]
Looks good!

File: src/simulation/wire_types/population.proto (hunk 2 above)
Comment (Janekdererste) [reply src/simulation/wire_types/population.proto:4]: We don't really need it, I just wanted to see, how one would do something like that. If we actually want to establish a binary file format, having a header with version information and maybe other stuff might be necessary.

**Comment by Janekdererste** [comment]
Two tests are failing but they don't fail locally...

**Pull request merged**